spec/controllers/v2/concerns/tailoring_file_spec.rb

```diff
@@ -0,0 +1,208 @@
+        os_major_version: 9,
+        ref_id_suffix: 'bar',
+        supports_minors: [8]
+      )
+    end
+
+    let(:item) do
+      FactoryBot.create(
+        :v2_tailoring,
+        :with_tailored_values,
+        policy: parent,
+        profile: canonical_profile,
+        os_minor_version: 8
+      )
+    end
+
+    it 'returns tailored values and default set of rules, but unselected' do
+      get :tailoring_file, params: extra_params.merge(parents: [:policy], format: :xml)
+
+      tailoring_file = Nokogiri::XML(response.body).remove_namespaces!
+      selected, deselected = tailoring_file.xpath('//Profile/select')
+                                           .partition { |sel| sel.attributes['selected'].value == 'true' }
+      values = tailoring_file.xpath('//Profile/set-value/@idref')
+
+      expect(response).to have_http_status :ok
+      expect(selected).to be_empty
+      expect(deselected).not_to be_empty
+      expect(deselected.map { |r| r.attributes['selected'].value }.uniq).to eq(['false'])
+      expect(values).not_to be_empty
+    end
+  end
+
+  context 'with randomly distributed rules' do
+    let(:canonical_profile) do
+      FactoryBot.create(
+        :v2_profile,
+        rule_count: 5,
+        value_count: 5,
+        security_guide: FactoryBot.create(:v2_security_guide, rule_count: 3, os_major_version: 9),
+        os_major_version: 9,
+        ref_id_suffix: 'bar',
+        supports_minors: [8]
+      )
+    end
+
+    let(:item) do
+      FactoryBot.create(
+        :v2_tailoring,
+        :with_mixed_rules, # tailor random rules
+        :with_tailored_values, # tailor random values
+        policy: parent,
+        profile: canonical_profile,
+        os_minor_version: 8
+      )
+    end
+
+    it 'returns XCCDF tailoring file' do
+      get :tailoring_file, params: extra_params.merge(parents: [:policy], format: :xml)
+
+      tailoring_values = V2::Tailoring.find(extra_params[:id]).value_overrides.keys
+      tailoring_file = Nokogiri::XML(response.body).remove_namespaces!
+      tailored_values = tailoring_file.xpath('//Profile/set-value/@idref')
+      selected, = tailoring_file.xpath('//Profile/select')
+                                .partition { |sel| sel.attributes['selected'].value == 'true' }
+      rules_and_groups = selected.map { |r| r.attributes['idref'].value }
+      rules_only = rules_and_groups.select { |r| r.exclude?('content_rule_group') }
+
+      expect(response).to have_http_status :ok
+      expect(response.headers['Content-Type']).to eq('application/xml')
+      expect(tailoring_values).to eq(tailored_values.map(&:value))
+      expect(rules_only).to match_array(item.rules_added.map(&:ref_id))
+      expect(rules_and_groups - rules_only).to match_array(item.rule_group_ref_ids)
+    end
+  end
+
+  context 'with unauthorized policy' do
+    let(:canonical_profile) do
+      FactoryBot.create(:v2_profile)
+    end
```

Comment:
Short enough to be a one-liner

```suggestion
    let(:canonical_profile) { FactoryBot.create(:v2_profile) }
```